src/Workspaces/Core/Portable/NamingStyles/NamingStyleRules.cs

```diff
@@ -46,5 +59,47 @@ private bool IsSymbolNameAnalyzable(ISymbol symbol)
 
             return true;
         }
+
+        private bool DoesSymbolImplementAnotherSymbol(ISymbol symbol)
+        {
+            if (symbol.IsStatic || symbol.DeclaredAccessibility != Accessibility.Public)
```

Comment:
The checks are out of order. Good catch. The public check should be after the IsOverride check / explicit implementation check. Will add a test for this as well.